hopp/simulation/technologies/resource/nsrdb_data.py

```diff
@@ -0,0 +1,226 @@
+from rex import NSRDBX
+from rex.sam_resource import SAMResource
+import numpy as np 
+from hopp.simulation.technologies.resource.resource import Resource
+from typing import Optional, Union
+from pathlib import Path
+import os
+from hopp.utilities.validators import range_val
+NSRDB_DEP = "/datasets/NSRDB/deprecated_v3/nsrdb_"
+
+# NOTE: Current version of PSM v3.2.2 which corresponds to /api/nsrdb/v2/solar/psm3-2-2-download 
+NSRDB_NEW = "/datasets/NSRDB/current/nsrdb_"
+
+# Pull Solar Resource Data directly from NSRDB on HPC
+# To be called instead of SolarResource from hopp.simulation.technologies.resource
+class HPCSolarData(Resource):
+    """
+    Class to manage Solar Resource data from NSRDB Datasets.
+
+    Attributes:
+        nsrdb_file: (str) path of file that resource data is pulled from.
+        site_gid: (int) id for NSRDB location that resource data was pulled from.
+        nsrdb_latitude: (float) latitude of NSRDB location corresponding to site_gid.
+        nsrdb_longitude: (float) longitude of NSRDB location corresponding to site_gid.
+        
+    """
+
+
+    def __init__(
+        self,
+        lat: float,
+        lon: float,
+        year: int,
+        nsrdb_source_path: Union[str,Path] = "",
+        filepath: str = "",
+        ):
+        """Class to pull solar resource data from NSRDB datasets hosted on the HPC
+
+        Args:
+            lat (float): latitude corresponding to location for solar resource data
+            lon (float): longitude corresponding to location for solar resource data
+            year (int): year for resource data. must be between 1998 and 2022
+            nsrdb_source_path (Union[str,Path], optional): directory where NSRDB data is hosted on HPC. Defaults to "".
+            filepath (str, optional): filepath to NSRDB h5 file on HPC. Defaults to "".
+                - should be formatted as: /path/to/file/name_of_file.h5
+        Raises:
+            ValueError: if year is not between 1998 and 2022 (inclusive)
+            FileNotFoundError: if nsrdb_file is not valid filepath
+        """
+       
+        # NOTE: self.data must be compatible with PVWatts.SolarResource.solar_resource_data
+        # see: https://nrel-pysam.readthedocs.io/en/main/modules/Pvwattsv8.html#PySAM.Pvwattsv8.Pvwattsv8.SolarResource
+        super().__init__(lat, lon, year)
+
+        if filepath == "" and nsrdb_source_path=="":
+            # use default filepath
+            self.nsrdb_file = NSRDB_NEW + "{}.h5".format(self.year)
```

Comment:
I'd recommend the use of f-strings in cases like this so it reads a bit easier. `f"{self.year}.h5"`. If you decide to make `NSRDB_NEW` a path, this becomes `NSRDB_NEW / f"{self.year}.h5"` as an example